### PR TITLE
Nadiaagahozo

### PR DESCRIPTION
**Pull request by nadiaagahozo**

### CODE REVIEW

File: app/models/skill.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class Skill < ApplicationRecord
+  belongs_to :student
+  validates :skill_name, presence: true
```
Comment (TziporahW) [inline app/models/skill.rb:3]: name not skill name

File: app/controllers/api/capstones_controller.rb [hunk 2]
```diff
@@ -1,2 +1,43 @@
 class Api::CapstonesController < ApplicationController
+  before_action :authenticate_student, except: [:index, :show]
+  def index
+    @capstones = Capstone.all
+    render "index.json.jb"
+  end
+
+  def create
+    @capstone = Capstone.new(
+      student_id: params[:current_student_id],
```
Comment (thewendels) [inline app/controllers/api/capstones_controller.rb]: student_id: current_student.id,

File: app/controllers/api/capstones_controller.rb [hunk 3]
```diff
@@ -1,2 +1,43 @@
...
+    @capstone.description = params[:description] || @capstone.description
+    @capstone.url = params[:url] || @capstone.url
+    @capstone.screenshot_url = params[:screenshot_url] || @capstone.screenshot_url
+    render "show.json.jb"
+  end
+  
+  def destroy
+    capstone = Capstone.find_by(id: params[:id])
+    capstone.destroy
+    render json: {message: "Photo successfully destroyed."}
```
Comment (thewendels) [inline app/controllers/api/capstones_controller.rb]: Capstone successfully destroyed

File: app/controllers/api/skills_controller.rb [hunk 4]
```diff
@@ -1,2 +1,39 @@
 class Api::SkillsController < ApplicationController
+  before_action :authenticate_student, except: [:index, :show]
+  def index
+    @skills = Skill.all
+    render "index.json.jb"
+  end
+
+  def create
+    @skill = Skill.new(
+      current_student_id: params[:current_student_id],
```
Comment (thewendels) [inline app/controllers/api/skills_controller.rb]: student_id: current_student.id,

File: app/controllers/api/skills_controller.rb [hunk 5]
```diff
@@ -1,2 +1,39 @@
...
+    @skill = skill.find_by(id: params[:id])
+    @skill.current_student_id = params[:current_student_id] || @skill.current_student_id
+    @skill.name = params[:name] || @skill.name
+    render "show.json.jb"
+  end
+  
+  def destroy
+    skill = Skill.find_by(id: params[:id])
+    skill.destroy
+    render json: {message: "Photo successfully destroyed."}
```
Comment (thewendels) [inline app/controllers/api/skills_controller.rb]: Skill successfully destroyed

File: app/controllers/api/skills_controller.rb [hunk 6]
```diff
@@ -1,2 +1,39 @@
...
+  before_action :authenticate_student, except: [:index, :show]
+  def index
+    @skills = Skill.all
+    render "index.json.jb"
+  end
+
+  def create
+    @skill = Skill.new(
+      current_student_id: params[:current_student_id],
+      name: params[:name],
```
Comment (thewendels) [inline app/controllers/api/skills_controller.rb]: skill_name rather than name

File: app/views/api/capstones/show.json.jb [hunk 7]
```diff
@@ -0,0 +1 @@
+render partial: 'captstone.json.jb', locals: { captstone: @captstone }
```
Comment (thewendels) [inline app/views/api/capstones/show.json.jb]: capstone is misspelled in every instance

**Pull request closed without merging**